examples/gin-example/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"log"
+	"net/http"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware/v2"
+	"github.com/auth0/go-jwt-middleware/v2/validator"
+	"github.com/gin-gonic/gin"
+)
+
+// Try it out with:
+//
+// eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnby1qd3QtbWlkZGxld2FyZS1leGFtcGxlIiwiYXVkIjoiYXVkaWVuY2UtZXhhbXBsZSIsInN1YiI6IjEyMzQ1Njc4OTAiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjIsInVzZXJuYW1lIjoidXNlcjEyMyJ9.XFhrzWzntyINkgoRt2mb8dES84dJcuOoORdzKfwUX70
+//
+// which is signed with 'secret' and has the data:
+//
+//	{
+//	  "iss": "go-jwt-middleware-example",
+//	  "aud": "audience-example",
+//	  "sub": "1234567890",
+//	  "name": "John Doe",
+//	  "iat": 1516239022,
+//	  "username": "user123"
+//	}
+//
+// You can also try out the custom validation with:
+//
+// eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnby1qd3QtbWlkZGxld2FyZS1leGFtcGxlIiwiYXVkIjoiYXVkaWVuY2UtZXhhbXBsZSIsInN1YiI6IjEyMzQ1Njc4OTAiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjIsInVzZXJuYW1lIjoidXNlcjEyMyIsInNob3VsZFJlamVjdCI6dHJ1ZX0.Jf13PY_Oyu2x3Gx1JQ0jXRiWaCOb5T2RbKOrTPBNHJA
+//
+// which is signed with 'secret' and has the data:
+//
+//	{
+//	  "iss": "go-jwt-middleware-example",
+//	  "aud": "audience-example",
+//	  "sub": "1234567890",
+//	  "name": "John Doe",
+//	  "iat": 1516239022,
+//	  "username": "user123",
+//	  "shouldReject": true
+//	}
+
+func main() {
+	router := gin.Default()
+	router.GET("/", checkJWT(), func(ctx *gin.Context) {
+		claims, ok := ctx.Request.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+		if !ok {
+			ctx.AbortWithStatusJSON(
+				http.StatusInternalServerError,
+				map[string]string{"message": "Failed to get validated JWT claims."},
+			)
+			return
+		}
+
+		customClaims, ok := claims.CustomClaims.(*CustomClaimsExample)
+		if !ok {
+			ctx.AbortWithStatusJSON(
+				http.StatusInternalServerError,
+				map[string]string{"message": "Failed to cast custom JWT claims to specific type."},
+			)
+			return
+		}
+
+		if len(customClaims.Username) == 0 {
```

Comment:
Should this be in `CustomClaimsExample.Validate()`?

Reply:
Not necessarily, I think the idea here was just to show that once the claims are validated, we can have extra business logic inside a specific handler to make use of them.